package.json

```diff
@@ -36,7 +43,10 @@
     "rollup-plugin-babel": "^4.1.0",
     "rollup-plugin-commonjs": "^10.0.0",
     "rollup-plugin-node-resolve": "^5.0.0",
-    "rollup-plugin-terser": "^4.0.4"
+    "rollup-plugin-terser": "^4.0.4",
+    "rollup-plugin-typescript": "^1.0.1",
```

Comment:
i think this may be the wrong plugin to use (https://www.npmjs.com/package/rollup-plugin-typescript)

the recommended version is https://www.npmjs.com/package/@rollup/plugin-typescript... maybe it will also solve the compile issue you mention above? (https://www.npmjs.com/package/@rollup/plugin-typescript#noemitonerror)